examples/ObjFile.cpp

```diff
@@ -0,0 +1,351 @@
+#include <FastBVH/BVH.h>
+#include <FastBVH/Traverser.h>
+
+#include <algorithm>
+#include <vector>
+
+#include <cstdio>
+#include <cstdlib>
+
+#include "Log.h"
+#include "SimpleScheduler.h"
+#include "Stopwatch.h"
+
+#include "tiny_obj_loader.h"
+
+namespace {
+
+//! \brief A single triangular face from a mesh.
+//! The positions and UV coordinates are indexed.
+//! In this example, an instance of 'Face' is a geometric primitive.
+struct Face final {
+  //! The indices to each position of the face.
+  int pos[3];
+  //! The indices to each UV coordinate.
+  int uv[3];
+};
+
+//! \brief Converts a face into a bounding box.
+//! Used by @ref FaceBVH::BVH during the build process.
+class FaceBoxConverter final {
+  /// The minimum X values of the mesh faces.
+  std::vector<float> min_x;
+  /// The minimum Y values of the mesh faces.
+  std::vector<float> min_y;
+  /// The minimum Z values of the mesh faces.
+  std::vector<float> min_z;
+  /// The maximum X values of the mesh faces.
+  std::vector<float> max_x;
+  /// The maximum Y values of the mesh faces.
+  std::vector<float> max_y;
+  /// The maximum Z values of the mesh faces.
+  std::vector<float> max_z;
```

Comment:
Agree that storing bounding boxes can work in this way, but seems like it will lead to bad cache behavior (necessary values for one face are scattered in memory rather than being contiguous). Alternatively, is it better to compute these from the triangles themselves during intersection to avoid looking these up? This is example code anyway, so I'm fine either way -- After profiling later, could update this part.

Reply:
Initially, this was done by making the calculations on the fly. Storing the min/max lthis way lead to a speed up going from 90ms of construction time to about 70ms (using the Sponza model.) The idea here is to use SoA to encourage auto vectorization. I think profiling or disassembly would be a good idea because I haven't yet checked to see if the loop is being vectorized or not. Definitely worth messing around with.